javascript/code-challenges/challenges-09.test.js

```diff
@@ -0,0 +1,322 @@
+CHALLENGE 4
+
+You are given an object with names and their coresponding phone numbers that looks like this:
+{
+  'Grace Hopper': '222-303-5938',
+  'Ada Lovelace': '222-349-9842',
+  'Alan Turing': '222-853-5933'
+}
+
+HR has asked you to change the data to make it easier to print so that it looks like this:
+[
+  'Grace Hopper: 222-303-5938',
+  'Ada Lovelace: 222-349-9842',
+  'Alan Turing: 222-853-5933'
+]
+
+------------------------------------------------------------------------------------------------ */
+
+const updateNumbers = (obj) => {
+  let printableData = [];
+  let objKeys = Object.keys(obj);
+  let objVals = Object.values(obj);
+  for (let i = 0; i < objKeys.length; i++) {
+    printableData.push((
+      objKeys[i] + ': ' + objVals[i])
+    );
+  }
+  return obj = printableData;
+};
+
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named getHouses that returns a new array containing the names of all of the houses in the data set.
+------------------------------------------------------------------------------------------------ */
+
+const characters = [
+  {
+    name: 'Eddard',
+    spouse: 'Catelyn',
+    children: ['Robb', 'Sansa', 'Arya', 'Bran', 'Rickon'],
+    house: 'Stark',
+  },
+  {
+    name: 'Jon',
+    spouse: 'Lysa',
+    children: ['Robin'],
+    house: 'Arryn',
+  },
+  {
+    name: 'Cersei',
+    spouse: 'Robert',
+    children: ['Joffrey', 'Myrcella', 'Tommen'],
+    house: 'Lannister',
+  },
+  {
+    name: 'Daenarys',
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen',
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell',
+  },
+  {
+    name: 'Sansa',
+    spouse: 'Tyrion',
+    house: 'Stark',
+  },
+  {
+    name: 'Jon',
+    spouse: null,
+    house: 'Snow',
+  },
+];
+
+const getHouses = (arr) => {
+  let houses = [];
+  arr.forEach(character => {
+    houses.push(character.house);
+  });
+  return houses;
+};
+
+/*------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named hasChildrenValues that uses Object.values to determine if any given character in the data set has children.
+
+This function should take in an array of data and a character name and return a Boolean.
+
+For example:
+hasChildrenValues(characters, 'Cersei') will return true
+hasChildrenValues(characters, 'Sansa') will return false
+------------------------------------------------------------------------------------------------ */
+
+const hasChildrenValues = (arr, character) => {
+  // let childCheckArr = [];
+  // arr.forEach(char => {
+  //   if (char.name === character) {
+  //     childCheckArr = Object.value(char);
+  //     if (childCheckArr[2].length > 0) {
+  //       return true;
+  //     }
+  //     return;
+  //   }
+  //   return;
```

Comment:
Really good work on all these!

I think you had  mentioned in your retro you had some questions on this one. I will try and help 👍 

- So for this one, you have the right idea here. We need to iterate over the arr of given characters.
- We kind of have to do two things here though:
  - The first check you have on line 156 is correct
  - The second check involves using Object.values(). 
  - Immediately after line 156 we can do another if statement, using the Object.values(). This second check though in the second if statement should check for children. 
  - We can remove lines 157-161
  - I used a for of loop for this and it was helpful 👍